framework/src/main/java/org/checkerframework/framework/stub/JavaStubifier.java

```diff
@@ -0,0 +1,184 @@
+    /**
+     * Processes each provided command-line argument.
+     *
+     * @param args command-line arguments: directories to process
+     */
+    public static void main(String[] args) {
+        if (args.length < 1) {
+            System.err.println("Usage: provide one or more directory names to process");
+            System.exit(1);
+        }
+        for (String arg : args) {
+            process(arg);
+        }
+    }
+
+    /**
+     * Process the given directory.
+     *
+     * @param dir directory to process
+     */
+    private static void process(String dir) {
+        Path root = Paths.get(dir);
+        MinimizerCallback mc = new MinimizerCallback();
+        ProjectRoot projectRoot = new ParserCollectionStrategy().collect(root);
+
+        projectRoot
+                .getSourceRoots()
+                .forEach(
+                        sourceRoot -> {
+                            try {
+                                sourceRoot.parse("", mc);
+                            } catch (IOException e) {
+                                System.err.println("IOException: " + e);
+                            }
+                        });
+    }
+
+    /** Callback to process each Java file. */
+    private static class MinimizerCallback implements SourceRoot.Callback {
+        /** The visitor instance. */
+        private final MinimizerVisitor mv;
+
+        /** Create a MinimizerCallback instance. */
+        public MinimizerCallback() {
+            this.mv = new MinimizerVisitor();
+        }
+
+        @Override
+        public Result process(
+                Path localPath, Path absolutePath, ParseResult<CompilationUnit> result) {
+            Result res = Result.SAVE;
+            // System.out.printf("Minimizing %s%n", absolutePath);
+            Optional<CompilationUnit> opt = result.getResult();
+            if (opt.isPresent()) {
+                CompilationUnit cu = opt.get();
+                // this somehow only removes comments except the
+                // first one, and copyright headers are kept
+                cu.getComments().forEach(Node::remove);
+                mv.visit(cu, null);
+                if (cu.findAll(ClassOrInterfaceDeclaration.class).isEmpty()
+                        && cu.findAll(AnnotationDeclaration.class).isEmpty()
+                        && cu.findAll(EnumDeclaration.class).isEmpty()
+                        && !absolutePath.endsWith("package-info.java")) {
+                    // All content is removed, delete this file.
+                    new File(absolutePath.toUri()).delete();
+                    res = Result.DONT_SAVE;
+                }
+            }
+            return res;
+        }
+    }
+
+    /** Visitor to processes one compilation unit. */
+    private static class MinimizerVisitor extends ModifierVisitor<Void> {
+        @Override
+        public ClassOrInterfaceDeclaration visit(ClassOrInterfaceDeclaration cid, Void arg) {
+            if (cid.isInterface()) {
```

Comment:
I had to add this; otherwise, the interface java files were just removed.  Methods in interfaces are public even if they are not marked as such.  An interface could have a default method whose body should be removed, so we may want to fix this some other way.

Reply:
The logic in `removeIfPrivateOrPkgPrivate` looks at the visibility of the interface itself and removes an interface if it is private or package private.
If the interface itself isn't visible, why should we keep it?
Which interface did you look at?

Reply:
`java.util.Map`, for example, is remove in MinimizerCallback#process because it has no methods marked as public or protected.  

 It's fine to remove the interface the interface is private or package private, which I suppose you can do by calling `removeIfPrivateOrPkgPrivate` before the return below.

Reply:
I've added some logic to handle implicitly public members. Map looks fine to me and private interfaces are still removed.

Reply:
Thanks!